types/src/test/java/org/infinispan/protostream/types/java/TypesMarshallingTest.java

```diff
@@ -69,23 +74,83 @@ public String toString() {
             '}';
    }
 
-   @Parameterized.Parameters
+   @Parameterized.Parameters(name = "{0}")
    public static Object[][] marshallingMethods() {
       return Arrays.stream(MarshallingMethodType.values())
             .flatMap(t -> switch (t) {
-               case BYTE_ARRAY, INPUT_STREAM, JSON -> Stream.of(new TestConfiguration(t, false, false, null));
+               case BYTE_ARRAY, INPUT_STREAM -> Stream.of(new TestConfiguration(t, false, false, null));
                default -> Stream.of(
                      new TestConfiguration(t, true, true, null),
                      new TestConfiguration(t, true, false, ArrayList::new),
                      new TestConfiguration(t, true, false, HashSet::new),
                      new TestConfiguration(t, true, false, LinkedHashSet::new),
                      new TestConfiguration(t, true, false, LinkedList::new),
+                     new TestConfiguration(t, true, false, LinkedList::new),
                      new TestConfiguration(t, true, false, TreeSet::new));
             })
             .map(t -> new Object[]{t})
             .toArray(Object[][]::new);
    }
 
+   @Test
+   public void testNullElement() throws IOException {
+      assumeTrue(testConfiguration.method != MarshallingMethodType.INPUT_STREAM && testConfiguration.method != MarshallingMethodType.BYTE_ARRAY);
+      testConfiguration.method.marshallAndUnmarshallTest(null, context, false);
+      testConfiguration.method.marshallAndUnmarshallTest(new WrappedMessage(null), context, false);
+      testConfiguration.method.marshallAndUnmarshallTest(Collections.singletonList(null), context, false);
+   }
+
+   @Test
+   public void testNestedWrappedMessage() throws IOException {
```

Comment:
The current JSON produced by this is:

```json
{"_type":"org.infinispan.protostream.WrappedMessage","wrappedTypeId":1005,"wrappedMessage":"GTBDX5fqKajeIaHZX/rYYGi9"}
```

Whereas I would expect the UUID type to be represented in a readable format.

I like the generic `marshallAndUnmarshallTest` but I think we need to make it more robust by also passing in the expected JSON output so that we can verify it's correct. I'm not sure how we do this in a way that satisfies all of the `MarshallingMethodType` values though :thinking:

Reply:
It was a matter of distinguishing when the field was coming from a wrapped message of from a primitive. This now outputs something like:

```json
{
   "_type":"org.infinispan.protostream.WrappedMessage",
   "wrappedMessage":{
      "_type":"org.infinispan.protostream.commons.UUID",
      "mostSigBitsFixed":16476212963255011726,
      "leastSigBitsFixed":11281785963689059814
   }
}
```

The tests in `ProtobufUtilTest` do something similar, validating the expected JSON, but it does not have as many tests as here. I'll create a new test dedicated to the JSON conversion and compare it against the contents of a file.

Reply:
I think the individual files are good for readability. AFAICT we're still missing test cases for prettyPrint? I don't think we need a test for all cases, but it would be good to have at least one just to be sure.

Reply:
Sounds like a good idea. I'll put together some nested objects.

Reply:
@ryanemerson, great suggestion! There was a problem where a raw list (`"list": ["a", "b"]`) didn't have the elements with proper indentation. This is fixed now.